### DISCUSSION TITLE
Scoped Extension methods

### DISCUSSION THREAD

**Discussion by bloodchild8906** [dotnet/csharplang#6314] [Language Ideas]
this would help alot having extension methods be scoped to a class/property avoid the work-arounds I commonly have to do

**Comment by huoyaoyuan** [comment]
Technically this is about limiting `using` directives to smaller ranges.

**Comment by bloodchild8906** [comment]
thats a very ugly solution I want to make a library that has extension methods that only apply to the classes I want thus making it easier to use and modify in the future
Michael Brown

.NET Developer

0840644884 (tel:0840644884)
